packages/paste-core/components/switch/src/SwitchGroup.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {InlineControlGroup} from '@twilio-paste/inline-control-group';
+import type {InlineControlGroupProps} from '@twilio-paste/inline-control-group';
+import {SwitchContext} from './SwitchContext';
+
+export interface SwitchGroupProps extends InlineControlGroupProps {
+  name: string;
+  onChange?: (checked: boolean) => void;
+}
+
+const SwitchGroup = React.forwardRef<HTMLFieldSetElement, SwitchGroupProps>(
+  (
+    {
+      children,
+      element = 'SWITCH_GROUP',
+      disabled = false,
+      errorText,
+      name,
+      onChange,
+      orientation = 'vertical',
+      i18nRequiredLabel = '(required)',
+      ...props
+    },
+    ref
+  ) => {
+    const onChangeHandler = React.useCallback(
+      (event: React.ChangeEvent<HTMLInputElement>): void => {
+        if (onChange != null) {
+          onChange(event.target.checked);
+        }
+      },
+      [onChange]
+    );
+
+    const contextValue = React.useMemo(() => {
```

Comment:
Because that would make it available to each Switch child, which doesn't necessarily need it I don't think. `required` is used in InlineControlGroup within just this component.